# <span style="color:#3306B0">The problem of world hunger and food waste</span>
[Based on the [2018 M3 Challenge problem](https://m3challenge.siam.org/resources/archives/2018-year-at-a-glance/2018-problem-better-ate-than-never-reducing-food-waste/) for high school students, and its [winning solutions](https://m3challenge.siam.org/resources/archives/2018-year-at-a-glance/2018-winning-solutions/)]

<P>&nbsp;</P>

<DIV ALIGN="CENTER">
    <IMG SRC="./food-image.jpg" width="50%"></IMG><BR></BR>
Image courtesy of <A HREF="https://m3challenge.siam.org/resources/archives/2018-year-at-a-glance/2018-problem-better-ate-than-never-reducing-food-waste/">M3 Challenge Contest</A>
</DIV>


According to the UN, the world currently produces enough food 
to feed everyone on the planet.  Yet, every day many millions 
of people are on the brink of starvation and/or malnutrition due 
to lack of access to food.  This paradoxical situation is caused 
by several intertwining factors, one of which is the enormous 
scale of wasted food in the world.  In a 2011 report, the Food and 
Agriculture Organization of the UN states that "roughly one-third 
of food produced for human consumption is lost or wasted globally, 
which amounts to 1.3 billion tons per year."

The tragic consequences of wasted food are that it contributes not 
only  to global hunger, but also increases a host of unnecessary 
environmental costs associated with producing, transporting, 
storing and disposing it.  Although food waste occurs in both 
high- and low-income countries, there are differences in sources 
and causes.  In general, waste occurs at various points 
throughout the food supply chain, from production to 
consumption.  In the United States, the Environmental Protection 
Agency estimates food is the largest single item of "trash" that 
enters landfills and incinerators.

In this lab we construct models to study the amount of food wasted 
at the consumption end, and explore ways in which wasted food can 
be repurposed to reduce world hunger.  Here is the problem statement, 
from the [M3 Challenge archives](https://m3challenge.siam.org/resources/archives/2018-year-at-a-glance/2018-problem-better-ate-than-never-reducing-food-waste/): 

<div class="alert alert-block alert-info">
<span style="color:#033600">
<ul>
  <li>Create a mathematical model that a state (in the United States) 
could use to determine if it could feed its food-insecure 
population using the wasted food generated in that state. Demonstrate 
how your model works for Texas; you may choose to use 
provided data. <BR>&nbsp;</BR>
  </li>
    <!--
  <li>Personal choices when it comes to food consumption primarily 
occur at the grocery store, school cafeteria, restaurants, and 
at home. Create a mathematical model that can be used to 
determine the amount of food waste a household generates in a year 
based on their traits and habits. Demonstrate how your model works 
by evaluating it for the following households (provided data may be 
helpful):
  <ul>
    <li>Single parent with a toddler, annual income of \$20,500 </li>
    <li>Family of four (two parents, two teenage children), 
    annual income of \$135,000  </li>
    <li>Elderly couple, living on retirement, annual income of \$55,000 </li>
    <li>Single 23-year-old, annual income of \$45,000 </li>
  </ul>
  </li>
    -->
</ul>
</span>
</div>

<div class="alert alert-block alert-info">
<span style="color:#033600">
Multiple agencies collect a wealth of data on a regular 
basis. A portion of this data has been compiled and provided. You 
are not required to use this data; that is, you may choose to 
use none, some, or all of this data and/or any additional data 
sources you may identify while working on this problem.  Be sure 
to cite all resources used. <BR>&nbsp;</BR>

[Texas_food_data](https://m3challenge.siam.org/wp-content/uploads/Texas_food_data.xlsx)
- This data may be useful for Question 1.
- The first tab has information about levels of food insecurity in TX.
- The second tab has information about food losses and waste in North America.
- The third tab has information about how much food is produced in TX.
- Data in these sheets was pulled from multiple sources, references to original data is prominent in each spreadsheet.

<!--
[FoodExpenditures](https://m3challenge.siam.org/wp-content/uploads/FoodExpenditures.xlsx)
- This data may be helpful to gain insight on food choices that 
people make. These tabs provide total U.S. food expenditures, 
including purchases by consumers, governments, businesses, and 
nonprofit organizations in 2014. 
- The first tab gives total dollar amount spent for food at home.
- The second tab gives the dollar amounts spent for food at home by different outlets.
- The third tab gives the total dollar amount spent for food away home.
- The fourth tab gives the dollar amounts spent for food away from home by different outlets.
- The fifth tab gives the food expenditures (at and away from home) by families and individuals as a share of disposable personal income .
- These tables were taken from larger sets found 
<A HREF="https://www.ers.usda.gov/data-products/food-expenditures/food-expenditures/#Food">here</A>.

[ConsumerBehaviorBasedonIncome](https://m3challenge.siam.org/wp-content/uploads/ConsumerBehaviorBasedon-Income.xlsx)
- This data (from a survey of about 130,000 consumers) may be helpful 
to gain insight on choices (including those about food and many 
other expenditures) that people make.
- Statistics are provided that describe the sample population.
- The data is split among different income levels and provides average annual expenditures with standard deviations in a wide range of categories.
- This data was taken from 
<A HREF="https://www.bls.gov/cex/tables.htm">here</A>.
- You can generate similar tailored spreadsheets 
<A HREF="https://www.bls.gov/cex/data.htm">here</A>. (Note:  You may 
need a Java-enabled internet browser to generate spreadsheets.  Do an 
internet search if you need assistance with finding a Java-enabled 
browser.)

[AverageDailyIntakeofFood](https://m3challenge.siam.org/wp-content/uploads/food_table1.xls)
- This data (from 2007-10 National Health and Nutrition 
Examination Surveys of 18,117 persons 1 year of age and older) may 
be helpful to gain insight on the amount of different food 
types that people eat.
- Food group types, as well as chosen units of measurement for 
each food type, are consistent with the USDA and US Health and 
Human Services publication, Dietary Guidelines for Americans 2010.
- This data was taken from 
<A HREF="https://www.ers.usda.gov/data-products/food-consumption-and-nutrient-intakes/">here</A>.
-->
</span>
</div>

## <span style="color:#336630">Model development</span> 
[Based on selected winning papers (https://m3challenge.siam.org/resources/archives/2018-year-at-a-glance/2018-winning-solutions/)

In the spirit of best practices, the M3 Challenge Contest recommends 
addressing each of the following components in the course of model 
development: clearly define the problem; state any assumptions 
made; identify the variables used;  construct the math model; 
analyze and assess the solutions.

### 1. Defining the problem

The goal of our problem may be stated as follows:

> We want to create a model to determine whether the amount of food wasted in any given state in the U.S. can be used to feed its food-insecure population.  The state of Texas will be used as example to demonstrate how the model works.
<!--
2. We also want to construct a model to determine the amount of food wasted per year by a household in the U.S. based on the traits and habits of its members.
-->

### 2. Strategy and assumptions


Before we begin, it is interesting to note that the winning 
teams arrived at completely different conclusions about Texas -- 
one team concluded the amount of wasted food is **not** 
sufficient to feed its food-insecure population, while another 
team concluded the opposite!  Yet, both teams were "right" and 
won prizes!  How can that happen?

A closer examination of the modeling strategy of these two teams 
reveals there were significant differences in the assumptions they 
made.  For example, food waste occurs at many different levels: 
individual households; supermarkets; cafes & restaurants; in 
warehouses; during transportation; in agriculture & harvesting, 
etc.  There are no clear guidelines on which type of food waste 
can be reused to feed people in need.  Teams 
made varying (but reasonable) assumptions on such questions, 
resulting in conclusions that were sometimes widely different.

The overall approach consists of estimating the 
amount (or dollar value) of food wasted in the state/province 
in one year, 
and comparing it with the amount of food needed to feed its 
food-insecure population for that year.  The estimate of food waste 
tends to differ significantly based on the assumptions 
made.  We will formulate a metric for food waste similar to that 
developed by Teams \#10494 and \#10917. The strategy involves 
making the following key assumptions

 1. All food wasted at any point in the production to consumption process will be considered usable for feeding the food-insecure.
 2. All wasted food can be transported and distributed to those in need without further waste, loss or damage.
 3. The dollar value of a food item is a reasonable proxy for the amount and/or quantity and/or calories of that item.
 4. All food-insecure individuals have the same food needs every day, and for the entire year.  Thus, we treat the food needs of children and adults the same.
 5. A state's cash receipts are a reasonable representation of all food purchased for consumption in that state.
 6. Average food waste percentages across North America are representative of the actual food waste in any specific state in the U.S.

While some of these assumptions may be hard to justify in a rigorous 
way (e.g., 1, 2 and 4), they offer a reasonable starting point for 
a process of iterative model development.  Thus, we will use them 
to build a first model.
 
 
 ### 3. The model and its variables

In order to determine whether there is enough food waste in a state to 
fulfill the needs of the food-insecure, we will formulate separate 
metrics for the annual waste and the annual needs, in units of US dollars ($W$ and $N$, 
respectively).  Then, we can define the net surplus as: $S = W-N$.  If 
$S$ is positive, then the food waste is sufficient to meet the 
needs of the food-insecure.  If $S$ is negative, then there is a 
deficit, which would mean the food waste is not sufficient 
for meeting the needs.  For practical implementation, we will make 
the following additional refinements, which are based on how the 
food data are sourced and organized:

- All food items will be classified into one of 5 categories: Protein, produce (fruits, vegetables, etc), grain, dairy, oil.
- Wasted food will be divided into two groups: Production waste, and consumption waste.

The "Venn diagram" style sketch below is helpful for conceptualizing 
the processes we will consider in our model.  The outermost 
region (light blue) represents the totality of food produced, 
including food wasted in production.  The inner grey region 
is the subset of food purchased for consumption, which is assumed 
to correspond to the cash receipts.  The innermost subset shown 
in white represents the food that is actually consumed.  In other 
words, everything outside the white region is wasted food, either 
in the production/distribution stage, or in the consumption 
stage.  Recall, these are all in units of US dollars.


<DIV ALIGN="CENTER">
<IMG SRC="./foodwaste_sketch.png"></IMG>
</DIV>

#### Metric for wasted food

General data about the percentage of food loss/waste in North America 
are available in the file 
[Texas_food_data.xlsx](https://m3challenge.siam.org/wp-content/uploads/Texas_food_data.xlsx).  In addition, the file contains data on 
the annual cash receipts in the state of Texas for all major food categories.  Using 
these data, we can formulate the annual food waste metric as follows <BR>&nbsp;

<span style="color:#830600">
$$
\begin{eqnarray*}
  W & = & \mbox{consumption waste} + \mbox{production waste} \\
  \mbox{consumption waste} & = &(\mbox{cash receipts}) \times 
      \left(\frac{\mbox{\% consumption waste}}{100}\right) ~~~~~~~~~~~ (1)
\end{eqnarray*}
$$
</span>

To compute production waste we use an indirect method, because 
the data only gives percentage waste, without any baseline 
figure.  Accordingly, we use the following two expressions 
and solve for production waste

$$
\begin{eqnarray*}
  \mbox{production waste} & = & (\mbox{total production})\cdot 
    (\mbox{\% production waste}/100)  \\
  \mbox{total production} & = & (\mbox{production waste}) +
      (\mbox{cash receipts})
\end{eqnarray*}
$$

The result is

<span style="color:#830600">
$$
  \mbox{production waste} = 
      \frac{(\mbox{cash receipts})\cdot(\mbox{\% production waste}/100)}
      {1 - (\mbox{\% production waste}/100)} ~~~~~~~~~~~ (2)
$$
</span>

Equations (1)-(2) define $W$ in terms of data that is readily 
available to us.  We note, however, that the waste metric will be 
computed individually for each food typecash receipts, and the results 
added to estimate the overall value of $W$.  This is because 
the data indicates high variability in cash receipts and waste 
percentage across different food groups.


#### Metric for food needs

The overall approach for computing the annual needs metric 
in a state is as follows <BR>&nbsp;

<span style="color:#830600">
$$
  N = (\mbox{\# of food insecure persons in the state}) \times 
  (\mbox{annual cost of feeding 1 person})
  ~~~~~~~~~~~ (3)
$$
</span>

For the state of Texas, data in the file [Texas_food_data.xlsx](https://m3challenge.siam.org/wp-content/uploads/Texas_food_data.xlsx) 
indicates the number of food insecure persons in 2015 was $4.32$ 
million and the average cost of a meal was $\$2.59$.  If we assume 
each food insecure person requires $3$ meals per day, for $365$ 
days a year, the annual cost of feeding 1 person 
$= \$2.59\times 3\times 365 = \$2,836.05$.  Therefore, for Texas

$$
  N = \$2,836.05 \times 4.32 \mbox{ million} = \$12,252 \mbox{ million}
$$

However, this total dollar figure is not very useful for making 
comparisons with the amount of wasted food, since there are 
significant disparities between needs and waste for different food 
groups.  What we would like is a breakdown of $N$ across the 
food categories, e.g., $N_\mbox{protein}, ~ N_\mbox{produce},~\ldots$ 
etc.  Then we can determine whether the amount of wasted proteins 
is sufficient to meet the protein needs, and so on.

In order to determine an appropriate breakdown of $N$, we consider 
average food intake guidelines recommended by public health 
agencies.  Team \#10494 computed the following daily intake 
percentages of different food groups, 
based on guidelines published in a document by the 
US Department of Agriculture


| Food category | Percent of diet |
| :-------------| :---------------|
| protein  | 8.6 % |
| fruit, vegetables, roots, tubers | 51.4 % |
| grain | 4.3 % |
| dairy | 34.3 % |
| oil | 1.4 % |

We will split $N$ into different food groups according to the 
proportions in this table.  For example, 
$N_\mbox{protein}= \$12,252 \times 0.086 = \$1,053.7 
\mbox{ million}$.  Proceeding in this way, we get the following augmented 
table

| Food category | Percent of diet &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; | $N_\mbox{category}$ (million \$) |
| :-------------| :---------------| :-------------------|
| protein  | 8.6 % | 1,053.7 |
| produce | 51.4 % | 6,297.5 |
| grain | 4.3 % |  526.8 |
| dairy | 34.3 % | 4,202.4 |
| oil | 1.4 % | 171.6 |

The values of $N_\mbox{category}$ represent the annual needs for that 
category of food in the state of Texas.  We can now compute the 
surplus or deficit of wasted food.


#### Surplus/deficit computation for Texas

Using data in the file [Texas_food_data.xlsx](https://m3challenge.siam.org/wp-content/uploads/Texas_food_data.xlsx) 
we will compute $W_\mbox{category}$ for each of the 7 food 
waste categories listed: (1) `cereals`, (2) `roots & 
tubers`, (3) `oilseeds & pulses`, (4) `fruit & vegetables`, (5) `meat`, 
(6) `fish & seafood`, 
(7) `milk`.  For the purpose of determining surplus or deficit, 
we will match these categories with the five in the food needs metric 
by consolidating as follows:

- protein = `meat` + `fish & seafood`
- produce = `fruit & vegetables` + `roots & tubers`
- grain = `cereals`
- dairy = `milk`
- oil = `oilseeds & pulses`

It is straightforward enough to use a hand calculator for computing 
the waste metric for each category, although the data input may be 
cumbersome.  We prefer to use the Python code given below (which 
was largely generated by `ChatGPT`) to do the needed 
computations.  Note that the production waste percent shown 
in the code combines all the wastes that occurs prior to consumption.

In [2]:
# Setup the input data
categories = ['meat', 'fish', 'fruit & veggies', 'roots & tubers', 'grain', 'dairy', 'oil']

# Data for cash receipts is in units of 1000 US dollars
cash_receipts_data = {
    'meat': [8470005, 1835520, 361309, 337269, 191218, 63803, 5643],
    'fish': [17722],
    'fruit & veggies': [1196184, 72500, 57293, 50430, 37848, 34884, 30319, 20664, 15392, 10873, 10543, 10193, 9240, 8580, 7043, 6932, 6707, 5585],
    'roots & tubers': [113522, 18368],
    'grain': [480609, 297693, 128209, 9618, 3929],
    'dairy': [1848140],
    'oil': [109663, 94020, 439982, 12894]
}

production_waste_percent = {
    'meat': 13.5,
    'fish': 27.5,
    'fruit & veggies': 38,
    'roots & tubers': 52,
    'grain': 10,
    'dairy': 5.7,
    'oil': 18
}

consumption_waste_percent = {
    'meat': 11,
    'fish': 33,
    'fruit & veggies': 28,
    'roots & tubers': 30,
    'grain': 27,
    'dairy': 15,
    'oil': 4
}

# Compute the total cash receipts for each category
cash_receipts = {category: sum(values) for category, values in cash_receipts_data.items()}

# Function to compute the waste metric
def compute_waste_metric(cash_receipts, production_waste_pct, consumption_waste_pct):
    production_waste = (cash_receipts * (production_waste_pct / 100)) / (1 - (production_waste_pct / 100))
    consumption_waste = cash_receipts * (consumption_waste_pct / 100)
    total_waste = production_waste + consumption_waste
    return total_waste

# Compute the waste metric for each category
waste_metrics = {}

for category in categories:
    cash = cash_receipts[category]
    prod_waste_pct = production_waste_percent[category]
    cons_waste_pct = consumption_waste_percent[category]
    
    waste = compute_waste_metric(cash, prod_waste_pct, cons_waste_pct)
    waste_metrics[category] = waste

# Print the waste metrics
print(f"Estimated food wastes per year in the state of Texas:\n")
for category, waste in waste_metrics.items():
    print(f"Waste metric for {category.capitalize()} = {waste/1000:,.2f} million USD")

print(f"\nCombined categories for later use:")
print(f"Waste metric for protein = Meat+Fish = \
{(waste_metrics['meat']+waste_metrics['fish'])/1000:,.2f} million USD")
print(f"Waste metric for produce = fruit+veggies+roots+tubers = \
{(waste_metrics['fruit & veggies']+waste_metrics['roots & tubers'])/1000:,.2f} million USD")

Estimated food wastes per year in the state of Texas:

Waste metric for Meat = 2,997.21 million USD
Waste metric for Fish = 12.57 million USD
Waste metric for Fruit & veggies = 1,420.80 million USD
Waste metric for Roots & tubers = 182.45 million USD
Waste metric for Grain = 350.64 million USD
Waste metric for Dairy = 388.93 million USD
Waste metric for Oil = 170.39 million USD

Combined categories for later use:
Waste metric for protein = Meat+Fish = 3,009.78 million USD
Waste metric for produce = fruit+veggies+roots+tubers = 1,603.24 million USD


### <span style="color:#336630">4. Results and model assessment</span>

The table below summarizes the computed results for food waste 
and food needs in Texas

| Category &nbsp; &nbsp; &nbsp; &nbsp;| $W$  (million \$) &nbsp; &nbsp; &nbsp; &nbsp; | $N$ (million \$) &nbsp; &nbsp; &nbsp; &nbsp; | $W-N$ (million \$)|
| :-------------| :---------------| :-------------------| :----|
| protein  | 3,009.8 | 1,053.7 | 1,956.1 |
| produce | 1,603.2 | 6,297.5 | $-$4,694.3 |
| grain | 350.6 |  526.8 |  $-$176.2 |
| dairy | 388.9 | 4,202.4 |  $-$3813.5  |
| oil | 170.4 | 171.6 |  $-$1.2  |

It appears that the only category for which there is a surplus of 
wasted food is proteins.  All the other categories show a deficit, 
suggesting the food waste in those categories is not sufficient 
to meet the needs of the food insecure population as a whole.  

<div class="alert alert-block alert-success">
<span style="color:#030600">
The conclusion, therefore, is the amount of food wasted in Texas is 
not sufficient to fulfill the needs of the entire food insecure 
population.  There is enough protein waste to meet the 
needs of the entire population.  However, waste in the other 
categories can only fulfill the needs of a smaller subset of the 
food insecure individuals.  <br> &nbsp;

We note, however, that the assumptions we made earlier regarding 
the food insecure population were quite conservative in at least 
two ways: (i) over 60% of food insecure individuals in Texas are 
listed as eligible for various government food aid programs, 
but we assumed everyone has the same needs; (ii) some individuals 
are food insecure for shorter periods than a full year, but we assumed 
a full year of food needs for everyone.  If we adjust these assumptions, 
it is possible that the amount of food waste is sufficient to feed 
a much larger proportion of food insecure individuals.
</span>
</div>

As always, the key question that begs to be addressed is, 
how reliable are these conclusions?  Should we believe these 
results?  To answer these questions, we first note that our 
model in this situation is just a set of simple algebraic calculations 
based on a bunch of assumptions.  The primary source of uncertainty 
in the results is due to uncertainties in the assumptions
and in the input data.

As discussed in the conclusions above, we assumed all food insecure 
individuals have the same needs year round.  While this may 
suffice for a rough approximation, it clearly overestimates the 
value of $N$ for all food categories.  A more accurate estimate 
of $N$ could well show there is a surplus of wasted food in 
some other categories as well (e.g., oil and grain).  Another assumption 
we made is that all wasted food, at any point in the 
production to consumption cycle, can be used to fulfill 
the needs of the food insecure.  But, this is clearly not true 
in reality for many reasons, including the perishable nature of some 
food items, and the logistics of procuring and transporting 
them on time.  Making real-world refinements to this assumption will 
result in lower values of $W$, epecially in categories such as 
produce, meat and fish.

To summarize, the particular choice of assumptions we've made 
play a crucial role in shaping the results and our conclusions.  In 
an iterative process of model development, our next step would 
be to refine these assumptions and incorporate them into an 
improved model.

To explore the effect of uncertainties in the input data, we 
will carry out some sensitivity studies in which we will perturb 
the input values by a small amount and observe the effect on 
$W$.  This is done in the Python code below.  We don't need to 
do this for $N$, since its behavior is linear with respect to 
perturbations in either of its inputs (number of food insecure 
people, cost of feeding per person).  That means, an $x\%$ change in 
either of its inputs will produce an $x\%$ change in $N$.

In [9]:
# Load the needed libraries & assign nicknames
import numpy as np

# Define a function to compute the waste metric
def compute_waste_metric(cash_receipts, production_waste_pct, consumption_waste_pct):
    production_waste = (cash_receipts * (production_waste_pct / 100)) / (1 - (production_waste_pct / 100))
    consumption_waste = cash_receipts * (consumption_waste_pct / 100)
    total_waste = production_waste + consumption_waste
    return total_waste

# Define a function to perturb the inputs and compute sensitivity
def sensitivity_analysis(cash_receipts, production_waste_pct, consumption_waste_pct, perturbation=0.01):
    original_waste = compute_waste_metric(cash_receipts, production_waste_pct, consumption_waste_pct)
    
    # Perturb cash receipts
    perturbed_cash = cash_receipts * (1 + perturbation)
    perturbed_waste_cash = compute_waste_metric(perturbed_cash, production_waste_pct, consumption_waste_pct)
    
    # Perturb production waste percentage
    perturbed_prod_pct = production_waste_pct * (1 + perturbation)
    perturbed_waste_prod = compute_waste_metric(cash_receipts, perturbed_prod_pct, consumption_waste_pct)
    
    # Perturb consumption waste percentage
    perturbed_cons_pct = consumption_waste_pct * (1 + perturbation)
    perturbed_waste_cons = compute_waste_metric(cash_receipts, production_waste_pct, perturbed_cons_pct)
    
    # Compute percentage changes
    sensitivity_cash = (perturbed_waste_cash - original_waste) / original_waste * 100
    sensitivity_prod = (perturbed_waste_prod - original_waste) / original_waste * 100
    sensitivity_cons = (perturbed_waste_cons - original_waste) / original_waste * 100
    
    return sensitivity_cash, sensitivity_prod, sensitivity_cons


# Perform sensitivity computations for specified perturbation percent
perturbation = 0.1  # Example perturbation of 10%

for category in categories:
    cash = cash_receipts[category]
    prod_waste_pct = production_waste_percent[category]
    cons_waste_pct = consumption_waste_percent[category]

    sensitivity_cash, sensitivity_prod, sensitivity_cons = sensitivity_analysis(cash, prod_waste_pct, cons_waste_pct, perturbation)

    print(f"Sensitivity of '{category}' waste metric to 10% perturbation:")
    print(f"  Cash receipts: {sensitivity_cash:.2f}%")
    print(f"  Production waste percentage: {sensitivity_prod:.2f}%")
    print(f"  Consumption waste percentage: {sensitivity_cons:.2f}%")
    print()


Sensitivity of 'meat' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 6.89%
  Consumption waste percentage: 4.13%

Sensitivity of 'fish' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 7.67%
  Consumption waste percentage: 4.65%

Sensitivity of 'fruit & veggies' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 11.79%
  Consumption waste percentage: 3.14%

Sensitivity of 'roots & tubers' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 18.30%
  Consumption waste percentage: 2.17%

Sensitivity of 'grain' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 3.28%
  Consumption waste percentage: 7.08%

Sensitivity of 'dairy' waste metric to 10% perturbation:
  Cash receipts: 10.00%
  Production waste percentage: 3.06%
  Consumption waste percentage: 7.13%

Sensitivity of 'oil' waste metric to 10

#### Interpretation of sensitivity results

Since cash receipts are linearly related to $W$, we are not 
surprised to see that a $10\%$ change in its value causes a 
$10\%$ change in $W$ for all food categories.  As for 
other observations, it appears that changes in production waste 
cause larger changes in $W$ than changes in consumption 
waste, for most food categories.  In fact, a $10\%$ change in 
consumption waste percentage causes $W$ to change by less than 
$5\%$ in all categories, except 'grain' and 'dairy'.  Overall, these 
results suggest it is important to accurately estimate the values 
of (at least) the cash receipts and production waste percentage 
for all food categories, since they have a significant impact on 
the value of $W$.  Likewise, since the value of $N$ linearly 
depends on the total number of food insecure persons and the 
per person feeding cost, those variables are important to accurately 
estimate as well.